Bottom: 1bbe88171be0107e37013cf10d2e42d8fdff173b
Top:    662a040332195b6d06f0966668fb5f5999994db6
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-05 23:06:58 -0400

Fix ups to make mmc over spi work












---

diff --git a/arch/powerpc/boot/dts/dspeak01.dts b/arch/powerpc/boot/dts/dspeak01.dts
index 398a949..c50e9c6 100644
--- a/arch/powerpc/boot/dts/dspeak01.dts
+++ b/arch/powerpc/boot/dts/dspeak01.dts
@@ -166,9 +166,9 @@
 			interrupt-parent = <&mpc5200_pic>;
 
 			mmc-slot@0 {
-				compatible = "mmc-spi";
+				compatible = "linux,mmc-spi";
 				reg = <0>;
-				max-speed = <50000000>;
+				spi-max-frequency = <20000000>;
 				/* Unregulated slot. */
 				voltage-range = <3300 3300>;
 				/*gpios = <&sdcsr_pio 1 0   /*  WP */
diff --git a/drivers/mmc/host/mmc_spi.c b/drivers/mmc/host/mmc_spi.c
index 72f8bde..9827f32 100644
--- a/drivers/mmc/host/mmc_spi.c
+++ b/drivers/mmc/host/mmc_spi.c
@@ -1438,7 +1438,7 @@ static int __devexit mmc_spi_remove(struct spi_device *spi)
 
 static struct spi_driver mmc_spi_driver = {
 	.driver = {
-		.name =		"mmc_spi",
+		.name =		"mmc-spi",
 		.bus =		&spi_bus_type,
 		.owner =	THIS_MODULE,
 	},
diff --git a/drivers/of/of_mmc_spi.c b/drivers/of/of_mmc_spi.c
index f556266..8911af5 100644
--- a/drivers/of/of_mmc_spi.c
+++ b/drivers/of/of_mmc_spi.c
@@ -95,7 +95,7 @@ static int of_mmc_spi_add(struct device *dev)
 	const u32 *voltage_range;
 	int size;
 
-	if (!np || !of_device_is_compatible(np, "mmc-spi"))
+	if (!np || !of_device_is_compatible(np, "linux,mmc-spi"))
 		return NOTIFY_DONE;
 
 	oms = kzalloc(sizeof(*oms), GFP_KERNEL);
@@ -152,7 +152,7 @@ static int of_mmc_spi_del(struct device *dev)
 	struct device_node *np = dev->archdata.of_node;
 	struct of_mmc_spi *oms;
 
-	if (!np || !of_device_is_compatible(np, "mmc-spi") ||
+	if (!np || !of_device_is_compatible(np, "linux,mmc-spi") ||
 			!dev->platform_data)
 		return NOTIFY_DONE;
